conformance/lang/expressions/regexp/regexp_semantics.balt

```diff
@@ -0,0 +1,719 @@
+
+    any & readonly reg6 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg6 is string:RegExp); // @output true
+
+    anydata & readonly reg7 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg7 is string:RegExp); // @output true
+
+    UserDefUnionType reg8 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(reg8 is string:RegExp); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as function parameter.
+Labels: RegExp, function-type-descriptor
+
+function getRegExp(string:RegExp re) returns string:RegExp {
+    return re;
+}
+
+function init() {
+    string:RegExp reg1 = re `([A-D]+|[H-Z]*)|C{2,9}`;
+    io:println(getRegExp(reg1) == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    string:RegExp reg2 = getRegExp(re `([A-D]+|[H-Z]*)|C{2,9}`);
+    io:println(reg2 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    string:RegExp reg3 = getRegExp(re = re `([A-D]+|[H-Z]*)|C{2,9}`);
+    io:println(reg3 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as function defaultable parameter.
+Labels: RegExp, defaultable-params, function-type-descriptor
+
+function getRegExp(string:RegExp re = re `ABC`) returns string:RegExp {
+    return re;
+}
+
+function init() {
+    string:RegExp reg1 = getRegExp();
+    io:println(getRegExp(reg1) == re `ABC`); // @output true
+
+    string:RegExp reg2 = getRegExp(re `EFG`);
+    io:println(getRegExp(reg2) == re `EFG`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as record field.
+Labels: RegExp, module-type-defn, record-type-descriptor
+
+type ReRec1 record {
+    string:RegExp re1;
+    ReRec2 rec?;
+};
+
+type ReRec2 record {
+    string:RegExp re2;
+};
+
+function init() {
+    ReRec1 r1 = {re1 : re `([A-D]+|[H-Z]*)|C{2,9}`};
+    io:println(r1.re1 is string:RegExp); // @output true
+    io:println(r1.re1 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+
+    ReRec1 r2 = {re1 : re `ABC`, rec : {re2 : re `DEF`}};
+    io:println((<ReRec2>r2.rec).re2 is string:RegExp); // @output true
+    io:println((<ReRec2>r2.rec).re2 == re `([A-D]+|[H-Z]*)|C{2,9}`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as record field with defaultable value.
+Labels: RegExp, module-type-defn, record-type-descriptor
+
+type ReRec1 record {
+    string:RegExp re1 = re `ABC`;
+};
+
+function init() {
+    ReRec1 r1 = {};
+    io:println(r1.re1 is string:RegExp); // @output true
+    io:println(r1.re1 == re `ABC`); // @output true
+}
+
+Test-Case: output
+Description: Test regular expression as table field.
+Labels: RegExp, table-constructor-expr, table-type-descriptor
+
+type Foo record {
+    readonly int id;
+    string:RegExp re;
+};
+
+table<Foo> key(id) t = table [
+    {id : 1, re : re `ABC`},
+    {id : 2, re : re `DEF`}
+];
+
+function init() {
+    io:println(t.get(1).re is string:RegExp); // @output true
+    io:println(t.get(1).re == re `ABC`); // @output true
```

Comment:
Instead of using table langlib functions here,shall we do member access on table and check the results
```balllerina
    Foo t2 = <Foo>t[1];
    io:println(t2.'re is string:RegExp); // @output true
```
Let's add the respective labels too

Reply:
Let's do this in other places that has used the table langlibs as well

Reply:
Let's do this in other places that have used the table langlib functions as well